propagators/opentracing/opentracing_propagator.go

```diff
@@ -0,0 +1,152 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package opentracing
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"strings"
+
+	"go.opentelemetry.io/otel/baggage"
+
+	"go.opentelemetry.io/otel/propagation"
+	"go.opentelemetry.io/otel/trace"
+)
+
+const (
+	// Default OpenTracing Header names.
+	traceIDHeader = "ot-tracer-traceid"
+	spanIDHeader  = "ot-tracer-spanid"
+	sampledHeader = "ot-tracer-sampled"
+
+	otTraceIDPadding = "0000000000000000"
+
+	traceID64BitsWidth = 64 / 4 // 16 hex character Trace ID.
+)
+
+var (
+	empty = trace.SpanContext{}
+
+	errInvalidSampledHeader = errors.New("invalid OpenTracing Sampled header found")
+	errInvalidTraceIDHeader = errors.New("invalid OpenTracing traceID header found")
+	errInvalidSpanIDHeader  = errors.New("invalid OpenTracing spanID header found")
+	errInvalidScope         = errors.New("require either both traceID and spanID or none")
+)
+
+// OpenTracing propagator serializes SpanContext to/from OpenTracing Headers.
+type OpenTracing struct {
+}
+
+var _ propagation.TextMapPropagator = OpenTracing{}
+
+// Inject injects a context into the carrier as OpenTracing headers.
+// NOTE: In order to interop with systems that use the OpenTracing header format, trace ids MUST be 64-bits
+func (ot OpenTracing) Inject(ctx context.Context, carrier propagation.TextMapCarrier) {
+	sc := trace.SpanFromContext(ctx).SpanContext()
+
+	if sc.TraceID.IsValid() && sc.SpanID.IsValid() {
+		carrier.Set(traceIDHeader, sc.TraceID.String()[len(sc.TraceID.String())-traceID64BitsWidth:])
+		carrier.Set(spanIDHeader, sc.SpanID.String())
+	}
+
+	if sc.IsSampled() {
```

Comment:
Wondering whether there's any value on injecting the sampled part if the trace/span ids are invalid.

Reply:
Yeah, probably this whole if/else can move inside the ids-validated branch above.